gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/RpcBatch.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns.spi;
+
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ChangesListResponse;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ManagedZonesListResponse;
+import com.google.api.services.dns.model.Project;
+import com.google.api.services.dns.model.ResourceRecordSetsListResponse;
+
+import java.util.Map;
+
+/**
+ * An interface for the collection of batch operations.
+ */
+public interface RpcBatch {
+
+  /**
+   * An interface for batch callbacks.
+   */
+  interface Callback<T> {
+
+    /**
+     * This method will be called upon success of the batch operation.
+     */
+    void onSuccess(T response);
+
+    /**
+     * This method will be called upon failure of the batch operation.
+     */
+    void onFailure(GoogleJsonError googleJsonError);
+  }
+
+  /**
+   * Adds a call to "list zones" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addListZones(Callback<ManagedZonesListResponse> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "create zone" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addCreateZone(ManagedZone zone, Callback<ManagedZone> callback,
+      Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "get zone" to the batch with the provided {@code callback} and {@code options}.
+   * The zone to be retrieved is identified by {@code zoneName}.
+   */
+  void addGetZone(String zoneName, Callback<ManagedZone> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "get project" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addGetProject(Callback<Project> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "delete zone" to the batch with the provided {@code callback} and {@code
+   * options}.
```

Comment:
There aren't options for this method, and instead you could use "The zone to be deleted is identified by {@code zoneName}"